crates/re_arrow_store/benches/arrow2.rs

```diff
@@ -0,0 +1,229 @@
+const NUM_ROWS: usize = 10_000;
+#[cfg(not(debug_assertions))]
+const NUM_INSTANCES: usize = 100;
+
+// `cargo test` also runs the benchmark setup code, so make sure they run quickly:
+#[cfg(debug_assertions)]
+const NUM_ROWS: usize = 1;
+#[cfg(debug_assertions)]
+const NUM_INSTANCES: usize = 1;
+
+// ---
+
+fn estimated_size_bytes(c: &mut Criterion) {
+    let kind = ["primitive", "struct"];
+
+    for kind in kind {
+        let mut group = c.benchmark_group(format!(
+            "arrow2/erased_clone/{kind}/rows={NUM_ROWS}/instances={NUM_INSTANCES}"
+        ));
+        group.throughput(criterion::Throughput::Elements(NUM_ROWS as _));
+
+        fn generate_cells(kind: &str) -> Vec<DataCell> {
+            match kind {
+                "primitive" => (0..NUM_ROWS)
+                    .map(|_| DataCell::from_native(build_some_instances(NUM_INSTANCES).as_slice()))
+                    .collect(),
+                "struct" => (0..NUM_ROWS)
+                    .map(|_| DataCell::from_native(build_some_point2d(NUM_INSTANCES).as_slice()))
+                    .collect(),
+                _ => unreachable!(),
+            }
+        }
+
+        {
+            {
+                let cells = generate_cells(kind);
+                let total_instances = cells.iter().map(|cell| cell.num_instances()).sum::<u32>();
+                assert_eq!(total_instances, (NUM_ROWS * NUM_INSTANCES) as u32);
+
+                group.bench_function("cell/arc_erased", |b| {
+                    b.iter(|| {
+                        let cells = cells.clone();
+                        assert_eq!(
+                            total_instances,
+                            cells.iter().map(|cell| cell.num_instances()).sum::<u32>()
+                        );
+                        cells
+                    });
+                });
+            }
+
+            {
+                let cells = generate_cells(kind).into_iter().map(Arc::new).collect_vec();
+                let total_instances = cells.iter().map(|cell| cell.num_instances()).sum::<u32>();
+                assert_eq!(total_instances, (NUM_ROWS * NUM_INSTANCES) as u32);
+
+                group.bench_function("cell/wrapped_in_arc", |b| {
+                    b.iter(|| {
+                        let cells = cells.clone();
+                        assert_eq!(
+                            total_instances,
+                            cells.iter().map(|cell| cell.num_instances()).sum::<u32>()
+                        );
+                        cells
+                    });
+                });
+            }
+
+            {
+                let cells = generate_cells(kind);
+                let arrays = cells.iter().map(|cell| cell.as_arrow()).collect_vec();
+                let total_instances = arrays.iter().map(|array| array.len() as u32).sum::<u32>();
+                assert_eq!(total_instances, (NUM_ROWS * NUM_INSTANCES) as u32);
+
+                group.bench_function("array", |b| {
```

Comment:
that's really `array/full_copy`, no?

Reply:
no full copy here, this is the usual arrow path which ends at `arrow2::Buffer::clone()` which is an `Arc`